modules/Layout.js

```diff
@@ -259,12 +246,22 @@ Layout.prototype.render = function (l) {
         x,y+h-5,
         x,y+4
       ], bg = l.selected?g.theme.bgH:g.theme.bg2;
-    g.setColor(bg).fillPoly(poly).setColor(l.selected ? g.theme.fgH : g.theme.fg2).drawPoly(poly);
+      g.setColor(bg).fillPoly(poly).setColor(l.selected ? g.theme.fgH : g.theme.fg2).drawPoly(poly);
     if (l.col!==undefined) g.setColor(l.col);
-    if (l.src) g.setBgColor(bg).drawImage("function"==typeof l.src?l.src():l.src, l.x + 10 + (0|l.pad), l.y + 8 + (0|l.pad));
-    else g.setFont("6x8",2).setFontAlign(0,0,l.r).drawString(l.label,l.x+l.w/2,l.y+l.h/2);
+    if (l.src) g.setBgColor(bg).drawImage(
+      "function"==typeof l.src?l.src():l.src,
+      l.x + l.w/2,
+      l.y + l.h/2,
+      {scale: l.scale||undefined, rotate: Math.PI*0.5*(l.r||0)}
```

Comment:
Just so you know, setting a rotation does center the image, but it also knocks the rendering out of the 'fast path' so images will render mucg slower.

Layout is already slow to update I know, so probably this doesn't matter - but it we just need to be careful we don't make it so slow it's unusable.

Reply:
Thank you for your time reviewing this Gordon.
Yes, I realised it would slow it down, but it is convenient to center the images. In the interest of speed though I will remove the option, and prepare rotated images outside of the layout.
Unless there is a fast way of rotating an image by increments of 90 degrees? Or for this use, flipping by 180?

Reply:
No, I think what you've done is the best option - I thought I'd better point it out though. Probably the best bet is to extend the 'fast path' inside Espruino if this becomes an issue.

Reply:
Thanks Gordon, I'll leave it as it is then. At the moment it doesn't feel any slower than before, but they are quite small images (9px square and 24px square 1bpp).